### PR TITLE
Pruning options

### PR DESCRIPTION
**Pull request by live627**
This set of commits tries to make it easier for mods to add their own pruning options. The most reliable way I could think of was to amalgamate them into a serialised array. I've added to the installer and upgrader to reflect this.

Currently, in the weekly scheduled task, there's no way to add a  pruning parameter without editing the file. I was thinking of splitting each pruning task into different methods, probably protected. What is the lowest PHP supported? 5.1.x?

### CODE REVIEW

File: other/upgrade_2-1_postgresql.sql [hunk 1]
```diff
@@ -242,3 +242,37 @@ upgrade_query("
...
+/******************************************************************************/
+---# Pruning options to use our new formmat
+---{
+$request = upgrade_query("
+	SELECT value
+	FROM {$db_prefix}settings
+	WHERE variable = 'pruningOptions'");
+if (mysql_num_rows($request) != 0)
+{
+	list ($oldValue) = mysql_fetch_row($request);
```
Comment (joshuaadickerson) [inline other/upgrade_2-1_postgresql.sql]: mysql_fetch_row?

**Comment by joshuaadickerson** [comment]
Has this been posted or tracked? It is a good idea but I was wondering if there was any previous discussion about it.

One thing we should look out for are more places where we should change a delimited list of values to a serialized array. I would prefer to work with serialized arrays.

File: other/upgrade_2-1_postgresql.sql (hunk 1 above)
Comment (live627) [inline other/upgrade_2-1_postgresql.sql]: Copy and paste error... yup, must fix!

**Comment by live627** [comment]
> Has this been posted or tracked? It is a good idea but I was wondering if there was any previous discussion about it.

Not that I know of. The hook in the admin is what inspired me to write this.

**Comment by live627** [comment]
Is this going in 2.1 at all?

**Comment by emanuele45** [comment]
Sorry, I've left this behind while working on some other stuff... :-[ (also because it didn't merge automatically and I'm still a bit scared (after merging the theme) by manual merges...but in the end that was just a couple of details)

First thing it doesn't work for me...I'm using a bit of a messed up configuration, but even changing the option to a serialized array every time I save the settings they are converted back to a list of comma-separated values.

That said, I think you are right, the hook in the settings page with the current code is "not particularly useful", though even with your changes another hook would be necessary in order to really allow mods to use the already written SMF pruning code, otherwise a mod should create "yet another scheduled task" in order to deal with the pruning (not a big deal one way or another though).

I talked with Spuds too and he agrees this would be a first step that should be applied broadly in a consistent way, so it would probably be a nice "rewrite" that you could lead in 2.2.
In few days I'll try to make a bit of order here creating a release branch and a development branch so that development for 2.2 can continue/start (and that should answer your next question: there will be a 2.2? There are people interested in making it, so there will likely be a release).

**Comment by live627** [comment]
I wanna make the scheduled task a class. Much more powerful than a mere hook IMO. Although, in that case, a hook of sorts would be needed too load  the file. Currently, it's either pre_include or pre_load.

This can certainly wait for 2.2! I think I'd prefer it, as the class rewrite may be a bit much for 2.1 and it's already behind schedule.

Master should be release branch while active development could go to develop.

**Comment by emanuele45** [comment]
Yep, a rewrite at that moment (and in general for 2.1 since Spuds (and partially me...I'm still wondering if I have actually done anything for 2.1...)  took it) would be a bit to much.

> Master should be release branch while active development could go to develop.

Yes, the fact is that the story of that repo is...well a mess! :P

**Pull request closed without merging**